src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,227 @@
+    List<Task> tasks = new ArrayList<>();
+    try {
+      tasks = storage.load(fileName);
+    } catch (FileNotFoundException e) {
+      storage.save(tasks, fileName);
+    }
+    this.tasks = tasks;
+  }
+
+  /**
+   * Constructs a new TaskList with no tasks and stores it in the default location.
+   *
+   * @throws IOException            if the file cannot be found or created.
+   * @throws ClassNotFoundException if the file cannot be deserialized.
+   * @see Storage
+   */
+  public TaskList() throws IOException, ClassNotFoundException, ClassCastException {
+    List<Task> tasks = new ArrayList<>();
+    try {
+      tasks = storage.load();
+    } catch (FileNotFoundException e) {
+      storage.save(tasks);
+    }
+    this.tasks = tasks;
+  }
+
+  /**
+   * Returns a string representation of the tasks in the list.
+   *
+   * @return a string representation of the tasks in the list.
+   * @see Ui#print(String)
+   */
+  public String[] stringify() {
+    if (this.tasks.size() == 0) {
+      return new String[]{"No tasks found."};
+    }
+    String[] outputs = new String[this.tasks.size()];
+    for (int i = 0; i < this.tasks.size(); i++) {
+      outputs[i] = addOrdinal(i, this.tasks.get(i));
+    }
+    return outputs;
+  }
+
+  /**
+   * Creates a new {@link Task} from the given arguments and adds it to the list.
+   *
+   * @param arguments the arguments to be used to create the task.
+   * @return a string representation of the task that was added.
+   */
+  public String[] add(String[] arguments) {
+    Task newTask;
+    switch (arguments[0]) {
+      case "/todo" -> newTask = new ToDo(arguments[1]);
+      case "/deadline" -> {
+        String[] deadlineArgs = Parser.parseArgs(arguments[1], new String[]{" /by "});
+        newTask = new Deadline(deadlineArgs[0], deadlineArgs[1]);
+      }
+      case "/event" -> {
+        String[] eventArgs = Parser.parseArgs(arguments[1], new String[]{" /from ", " /to "});
+        newTask = new Event(eventArgs[0], eventArgs[1], eventArgs[2]);
+      }
+      default -> throw new IllegalArgumentException("Invalid task type: " + arguments[0]);
+    }
+    this.tasks.add(newTask);
+    this.save();
+    return new String[]{"added: " + newTask};
+  }
+
+  /**
+   * Marks (toggle) the task(s) at the given index(es).
+   *
+   * @param indStr the index(es) of the task(s) to be marked as done.
+   * @return a string representation of the task(s) that were (un)marked.
+   */
+  public String[] mark(String[] indStr) {
+    List<Integer> indexes = new ArrayList<>();
+    Consumer<Integer> mark = ind -> {
+      this.tasks.get(ind).mark();
+      indexes.add(ind);
+    };
+    this.consume(indStr, mark);
+    this.save();
+    if (indexes.size() == 0) {
+      throw new IllegalArgumentException("No tasks found.");
+    } else if (indexes.size() == 1) {
+      Task task = this.tasks.get(indexes.get(0));
+      return new String[]{(task.isDone() ? "marked: " : "unmarked: ")
```

Comment:
```suggestion
      return new String[] { (task.isDone() ? "marked: " : "unmarked: ")
```
I think for all of your arrays may need a whitespace for the `{` and `}` to comply the 'White space within a statement' standard